contracts/amm/pool/Joseph.sol

```diff
@@ -15,31 +15,37 @@ abstract contract Joseph is JosephInternal, IJoseph {
     using SafeCast for int256;
 
     function calculateExchangeRate() external view override returns (uint256) {
-        return _calculateExchangeRate(block.timestamp);
+        IMiltonInternal milton = _getMilton();
+        return
+            _calculateExchangeRate(
```

Comment:
Wouldn't it be better to pass soap instead of calculateTimestamp and milton?

Reply:
You right. I will try fix it.